### PR TITLE
Fix common spelling mistakes

### PR DESCRIPTION
**Pull request by matmair**
We have a lot of spelling mistakes in the codebase, as recent PRs showed. This PR:
- adds spell checking to the hook pipeline
- fixes existing mistakes so we have a clean running CI

### CODE REVIEW

**Comment by wolflu05** [comment]
Nice addition, was already using a vscode extension, but doing this as part of linting is the better solution.

**Comment by SchrodingersGat** [comment]
Wow quite a large number of errors 😕

File: InvenTree/stock/api.py [hunk 1]
```diff
@@ -364,7 +364,7 @@ class Meta:
         ]
 
     # Relationship filters
-    manufactuer = rest_filters.ModelChoiceFilter(label='Manufacturer', queryset=Company.objects.filter(is_manufacturer=True), field_name='manufacturer_part__manufacturer')
+    manufacturer = rest_filters.ModelChoiceFilter(label='Manufacturer', queryset=Company.objects.filter(is_manufacturer=True), field_name='manufacturer_part__manufacturer')
```
Comment (matmair) [inline InvenTree/stock/api.py:367]: @SchrodingersGat I do not think that the change in this filter will break anything - am I overseeing anything?

**Review by SchrodingersGat — APPROVED** [review]
Some *actual* bugs in here which the spellcheck has fixed, too...

**Comment by SchrodingersGat** [comment]
Thanks @matmair for looking into this :)

**Pull request merged**